code/5677-matea-mitrovska/script.js

```diff
@@ -1,3 +1,107 @@
 document.addEventListener('DOMContentLoaded', () => {
     // Initialize the application when the DOM is fully loaded
 });
+
+let episodes = [];
+let sortedField = null;
+let sortOrder = 1;
+
+const fetchData = async () => {
+    const response = await fetch('https://raw.githubusercontent.com/sweko/internet-programming-a98db973kwl8xp1lz94kjf0bma5pez8c/refs/heads/main/data/doctor-who-episodes.json');
+    const data = await response.json();
+    episodes = data.episodes;
+    setupFilters();
+    displayEpisodes();
+};
+
+const setupFilters = () => {
+    const eraFilter = document.getElementById('era-filter');
+    const doctorFilter = document.getElementById('doctor-filter');
+    const companionFilter = document.getElementById('companion-filter');
+    const eras = [...new Set(episodes.map(ep => ep.era))];
+    const doctors = [...new Set(episodes.map(ep => ep.doctor.actor))];
+    const companions = [...new Set(episodes.map(ep => ep.companion?.actor).filter(Boolean))];
+
+    eras.forEach(era => eraFilter.innerHTML += `<option value="${era}">${era}</option>`);
+    doctors.forEach(doc => doctorFilter.innerHTML += `<option value="${doc}">${doc}</option>`);
+    companions.forEach(comp => companionFilter.innerHTML += `<option value="${comp}">${comp}</option>`);
+};
+
+const displayEpisodes = () => {
+    const nameInput = document.getElementById('name-filter').value.toLowerCase();
+    const eraValue = document.getElementById('era-filter').value;
+    const doctorValue = document.getElementById('doctor-filter').value;
+    const companionValue = document.getElementById('companion-filter').value;
+    const episodeContainer = document.querySelector('.episodes-list');
+    episodeContainer.innerHTML = '<div class="episodes-header">' + episodeContainer.firstElementChild.innerHTML + '</div>';
```

Comment:
This line essentially recreates the header, thus destroying the sorting handlers 😞